packages/react/src/components/Modal/Modal.mdx

```diff
@@ -1,15 +1,325 @@
+  - [Component API](#component-api)
+  - [Opening/closing modal](#openingclosing-modal)
+  - [Modal sizes](#modal-sizes)
+    - [Alignment](#alignment)
+    - [Overflow content](#overflow-content)
+  - [Modal button variants](#modal-button-variants)
+  - [Using modal title as message](#using-modal-title-as-message)
+  - [Focus management](#focus-management)
+    - [Setting `data-modal-primary-focus` attribute to the target element](#setting-data-modal-primary-focus-attribute-to-the-target-element)
+    - [Specifying a query selector to the target element](#specifying-a-query-selector-to-the-target-element)
+  - [References](#references)
+  - [Feedback](#feedback)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+import { Story, Props } from '@storybook/addon-docs/blocks';
+import Modal from '../Modal';
+import { InlineNotification } from '../Notification';
 
 # Modal
 
+[Source code](https://github.com/carbon-design-system/carbon/tree/master/packages/react/src/components/Modal)
+&nbsp;|&nbsp;
+[Usage guidelines](https://www.carbondesignsystem.com/components/modal/usage)
+&nbsp;|&nbsp;
+[Accessibility](https://www.carbondesignsystem.com/components/modal/accessibility)
+
 ## Overview
 
+`<Modal>` allows you to use its bespoke set of contents. `children` prop maps to
+the modal body content. You can also use `modalLabel`, `modalHeading`,
+`secondaryButtonText` and `primaryButtonText` props to change the corresponding
+text.
+
+<Preview>
+  <Story id="modal--with-state-manager" />
+</Preview>
+
 ## Component API
 
 <Props />
 
+## Opening/closing modal
+
+For both modal variants, you can open/close the modal by changing the `open`
+prop. For example, you can implement a launcher button with a React state and a
+`<Button>` that changes the state:
+
+```jsx
+function ModalStateManager() {
+  const [open, setOpen] = useState(false);
+  return (
+    <>
+      {typeof document === 'undefined'
+        ? null
+        : ReactDOM.createPortal(
+            <ComposedModal open={open} onClose={() => setOpen(false)}>
+              ...
+            </ComposedModal>,
+            document.body
+          )}
+      <Button kind="primary" onClick={() => setOpen(true)}>
+        Open modal
+      </Button>
+    </>
+  );
+}
+```
+
+You can create an abstract version of such state manager, as shown below.
+
+```jsx
+const ModalStateManager = ({
+  renderLauncher: LauncherContent,
+  children: ModalContent,
+}) => {
+  const [open, setOpen] = useState(false);
+  return (
+    <>
+      {!ModalContent || typeof document === 'undefined'
+        ? null
+        : ReactDOM.createPortal(
+            <ModalContent open={open} setOpen={setOpen} />,
+            document.body
+          )}
+      {LauncherContent && <LauncherContent open={open} setOpen={setOpen} />}
+    </>
+  );
+};
+```
+
+`<ModalWrapper>` is a modal state manager implementation which ships out of the
+box with Carbon.
+
+<InlineNotification kind="warning">
```

Comment:
Can we perhaps have the `title` say something like `Warning`?

![Screen Shot 2020-09-25 at 10 28 53 AM](https://user-images.githubusercontent.com/11928039/94297808-f51ff900-ff19-11ea-842f-a19d81b90809.png)